tlatools/org.lamport.tlatools/src/tla2sany/drivers/SANY.java

```diff
@@ -313,20 +288,22 @@ public static void frontEndSemanticAnalysis(SpecObj spec,
           Generator gen = new Generator(externalModuleTable, semanticErrors);
     
           // Perform semantic analysis and create semantic graph for one external module here
-          moduleNode = gen.generate(syntaxTreeRoot);    
-                      
-          // Set the isStandard field of the moduleNode.
-          // Added by LL on 24 July 2013
-          moduleNode.setStandard(spec.getResolver().isStandardModule(moduleStringName)) ;
-          
-          // Put the semantic graph and related info for moduleNode into the module table
-          externalModuleTable.put(UniqueString.uniqueStringOf(moduleStringName), 
-                                  gen.getSymbolTable().getExternalContext(),
-                                  moduleNode);
-  
-          // Level check if semantic analysis succeeded and levelCheck is true.
-          if (moduleNode != null && semanticErrors.isSuccess() && levelCheck ) {
-            moduleNode.levelCheck(1);
+          Optional<ModuleNode> moduleNodeOpt = gen.generate(syntaxTreeRoot);
```

Comment:
gen.generate can return a null. Optional is a better way to handle them. Previous code would fail on line 320 with nullpointer